src/shogun/mathematics/linalg/BaseVector.h

```diff
@@ -54,7 +54,7 @@ struct BaseVector
 	}
 
 	/** Data Storage */
-	virtual bool onGPU() = 0 ;
+	virtual bool onGPU() const = 0 ;
```

Comment:
well instead of dynamic casting see the SGObject `get_name` virtual function. that's a more general way to identify the type of the object without dynamic casting.